### PR TITLE
Refactor `get_broadcasts`

### PR DESCRIPTION
**Pull request by bIgBV**
* Consider size of the vector when checking for size.
* Drop broadcasts which cannot be serialized.
* Add new test for ordering.

### CODE REVIEW

File: src/broadcast.rs [hunk 1]
```diff
@@ -166,16 +166,19 @@ impl TransmitQueue {
         self.add(limited_broadcast);
     }
 
-    /// Returns a list of `LimitedBroadcast`s ordered by newest first. The limit parameter limits
-    /// the total number of `LimitedBroadcast`s returned. The size being counted is the
-    /// `serialized_size` of the `Broadcast` type.
-    pub fn get_broadcasts(&mut self, limit: u64) -> Result<Vec<LimitedBroadcast>> {
+    /// Returns a list of `Broadcast`s ordered by newest first. The number of values returned
+    /// is limited by the `limit` parameter which limits the toal size of the resultant Vec.
+    pub fn get_broadcasts(&mut self, limit: u64) -> Result<Vec<Broadcast>> {
```
Comment (LucioFranco) [inline src/broadcast.rs]: just for clairity sake this should be limit_bytes and a usize.

File: src/broadcast.rs [hunk 2]
```diff
@@ -205,7 +208,16 @@ impl TransmitQueue {
             let mut prune = vec![];
 
             for item in self.set.range(start..end) {
-                let size = serialized_size(&item.broadcast)? as i64;
+                // Drop those broadcasts which cannot be serialized.
+                let size = match serialized_size(&item.broadcast) {
+                    Ok(n) => n as i64,
+                    Err(e) => {
+                        eprintln!("Serialization error: {:?}", e);
```
Comment (LucioFranco) [inline src/broadcast.rs]: you can uncomment the `tokio-trace` stuff in `lib` and use `error!()` here.

**Review by LucioFranco — APPROVED** [review]
🙇 this is awesome!!!!

File: src/broadcast.rs (hunk 2 above)
Comment (bIgBV) [reply src/broadcast.rs]: Oh yeah, since this is a library, we don't need to initialize anything right?

Comment (LucioFranco) [reply src/broadcast.rs]: correct

**Pull request closed without merging**